validator/src/main.rs

```diff
@@ -2666,91 +2768,88 @@ pub fn main() {
+                }
+            }
+            (false, false) => {}
+            _ => {
+                eprintln!("{}", error_msg);
+                exit(1);
+            }
         }
+    }
+    block_engine_config.trust_packets = matches.is_present("trust_block_engine_packets");
 
-        let addr: String = value_of(&matches, "relayer_url")
-            .unwrap_or_else(|| value_of(&matches, "relayer_address").expect("missing relayer-url"));
-        let mut backend_endpoint =
-            Endpoint::from_shared(addr.clone()).expect("invalid relayer-address value");
-        if addr.contains("https") {
-            backend_endpoint = backend_endpoint
-                .tls_config(tonic::transport::ClientTlsConfig::new())
-                .expect("failed to set tls_config");
+    let mut relayer_config = RelayerConfig::default();
+    if matches.is_present("relayer_url") {
+        relayer_config.relayer_url =
+            value_of(&matches, "relayer_url").expect("couldn't parse relayer_url");
+    } else {
+        let error_msg = "Specifying seperate auth and backend addresses for relayer is deprecated.  Recommended to use --relayer_url instead.\
+                    If using relayer_auth_service_address and relayer_address, they must both be provided and set to the same value.";
+        match (
+            matches.is_present("relayer_auth_service_address"),
+            matches.is_present("relayer_address"),
+        ) {
+            (true, true) => {
+                let auth_addr: String = value_of(&matches, "relayer_auth_service_address")
+                    .expect("couldn't parse relayer_auth_service_address");
+                let backend_addr: String =
+                    value_of(&matches, "relayer_address").expect("couldn't parse relayer_address");
+                if auth_addr != backend_addr {
+                    eprintln!("{}", error_msg);
+                    exit(1);
+                } else {
+                    relayer_config.relayer_url = backend_addr;
+                }
+            }
+            (false, false) => {}
+            _ => {
+                eprintln!("{}", error_msg);
+                exit(1);
+            }
         }
+    }
+    relayer_config.trust_packets = matches.is_present("trust_relayer_packets");
 
-        let expected_heartbeat_interval_ms =
-            value_of(&matches, "relayer_expected_heartbeat_interval_ms").unwrap_or(500);
-        let expected_heartbeat_interval = Duration::from_millis(expected_heartbeat_interval_ms);
-
-        let max_failed_heartbeats =
-            value_of(&matches, "relayer_max_failed_heartbeats").unwrap_or(3);
-        assert!(
-            max_failed_heartbeats > 0,
-            "relayer-max-failed-heartbeats must be greater than zero"
+    let expected_heartbeat_interval_ms: u64 =
+        value_of(&matches, "relayer_expected_heartbeat_interval_ms").unwrap_or(
+            DEFAULT_RELAYER_EXPECTED_HEARTBEAT_INTERVAL_MS
```

Comment:
i think you can set this as a default in the arguments?

Reply:
it is set there, will change this to just unwrap()